src/systems/camera-mirror.js

```diff
@@ -80,11 +80,15 @@ AFRAME.registerSystem("camera-mirror", {
       if (playerHead) {
         tempHeadScale.copy(playerHead.scale);
         playerHead.scale.set(1, 1, 1);
+        playerHead.updateMatrices(true, true);
+        playerHead.updateMatrixWorld(true, true);
```

Comment:
`updateMatrices` walks up the tree, then `updateMatrixWorld` walks down it.

`updateMatrixWorld` is definitely a textbook case of the negative impact of a poorly named function. you'd assume by reading it that it updates the world matrix of the object you call it on as its primary operation. in fact: 

- it does not update the world matrix correctly, because it assumes the parent's local + world matrices up the chain are up-to-date (they may not be if the parent has had its pos/rot/scale modified in this frame)
 
- updates the local matrix too (but only if it's auto updating)

- updates the entire subtree's world matrices as well (but only if the world matrix was stale)

`updateMatrixWorld` makes sense for the outer "update all the dirty world matrices" tick loop that three.js does, but it should have never been a public API. it's used in various pieces of code assuming that it's how you update the world matrix of the object you call it on.

the new monkey patch exposes `updateMatrices` for this use-case as a public API you can use to update the local and/or world matrices on the object you call it on. it will have no effect on the children of the object, and will ensure that the necessary parents' matrices are up-to-date (unless you force it to not do this work.)